python/pydantic_core/core_schema.py

```diff
@@ -2418,10 +2422,14 @@ def with_default_schema(
         metadata: Any other information you want to include with the schema, not used by pydantic-core
         serialization: Custom serialization schema
     """
+
+    has_arg = isfunction(default_factory) and len(signature(default_factory).parameters) > 0
```

Comment:
@nix010, this check should reside in `pydantic`, like we do for the validators and determining their info status 👍

Reply:
Got it!.

Reply:
@sydney-runkle is this func is what you've mentioned ?
https://github.com/pydantic/pydantic/blob/60f704721a096dced3b19dd465a1e31495fcd802/pydantic/json_schema.py#L1076.
I don't see where it got called in the process.
Also so do you think that this line should be removed in `pydantic-core` ?

Reply:
Yep, this line should be removed in `pydantic-core`, bc the inspect logic should reside in `pydantic`.

Re inspection, I'd refer to this as an example: https://github.com/pydantic/pydantic/blob/6c3d3b3bb255422fc64eec8e9014016743207ff0/pydantic/_internal/_decorators.py#L514-L548

Reply:
@sydney-runkle sorry, I took a while but can't find a good place to put this check in the `pydantic` and accomodate with this, so feel free to take over this one.

Reply:
@nix010 no worries, I'll take it over today